pkgs/build-support/trivial-builders.nix

```
@@ -477,7 +477,7 @@ rec {
       cd $out
       ${lib.concatMapStrings (x: ''
           mkdir -p "$(dirname ${lib.escapeShellArg x.name})"
-          ln -s ${lib.escapeShellArg x.path} ${lib.escapeShellArg x.name}
+          ln -s ${x.path} ${lib.escapeShellArg x.name}
```

```suggestion
          ln -s ${lib.escapeShellArg "${x.path}"} ${lib.escapeShellArg x.name}
```
otherwise this is unsafe with paths in derivations that contain shell special characters

---

Is this possible ? I thought store paths were shell safe and if I try, I get this error:

```
error: store path '2ali0qarp8ym6q67abkqiggdyli9ffvz-a b' contains illegal character ' '
(use '--show-trace' to show detailed location information)
```

This attr should only be a path, anything else wouldn't be accessible from the build script, this should be a checked.

---

maybe we misunderstood, but isn't it legal to set eg `path = "${somePkg}/lib/path with spaces"`? if that's not allowed (and all other metacharacters are likewise illegal) then plain interpolation would work fine, but the extra safety escaping provides for future changes would still be a good thing